dp/pipeline/epc_arp.c

```diff
@@ -1739,7 +1742,12 @@ init_netlink_socket(void)
 
 	addr_t.nl_family = PF_NETLINK;
 	addr_t.nl_pad = 0;
+#ifdef USE_AF_PACKET
+	/* avoid address re-bind errors (with mnl_sock) by resetting pid value */
+	addr_t.nl_pid = 0;
```

Comment:
Can we make 0 the default in all cases
http://man7.org/linux/man-pages/man7/netlink.7.html

```
       nl_pid is the unicast address of netlink socket.  It's always 0 if
       the destination is in the kernel.  For a user-space process, nl_pid
       is usually the PID of the process owning the destination socket.
       However, nl_pid identifies a netlink socket, not a process.  If a
       process owns several netlink sockets, then nl_pid can be equal to the
       process ID only for at most one socket.  There are two ways to assign
       nl_pid to a netlink socket.  If the application sets nl_pid before
       calling bind(2), then it is up to the application to make sure that
       nl_pid is unique.  If the application sets it to 0, the kernel takes
       care of assigning it.  The kernel assigns the process ID to the first
       netlink socket the process opens and assigns a unique nl_pid to every
       netlink socket that the process subsequently creates.
```

Reply:
Yes, we can set it to 0 by default in all cases. I think a better way to fix this will be to submit a pull request independent of the AF_PACKET patch.